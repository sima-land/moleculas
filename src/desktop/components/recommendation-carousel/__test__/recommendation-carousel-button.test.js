import React from 'react';
import CarouselButton from '../recommendation-carousel-button';
import { shallow } from 'enzyme';

describe('<CarouselButton />', () => {
  it('renders correct without props', () => {
    const button = shallow(<CarouselButton />);
    expect(button).toMatchSnapshot();
  });
  it('renders correct with backward type', () => {
    const button = shallow(<CarouselButton type='backward' />);
    expect(button).toMatchSnapshot();
  });
  it('calls onUse properly', () => {
    const onUseFn = jest.fn();
    const button = shallow(<CarouselButton onUse={onUseFn} />);
    expect(onUseFn).not.toHaveBeenCalled();
    button.simulate('click');
    expect(onUseFn).toHaveBeenCalledTimes(1);
  });
});
