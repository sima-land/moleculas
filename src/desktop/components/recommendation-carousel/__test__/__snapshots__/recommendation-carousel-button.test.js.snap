// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselButton /> renders correct with backward type 1`] = `
<button
  aria-label="назад"
  className="control backward"
>
  <Icon
    icon={[Function]}
    role="presentation"
    size={24}
  />
</button>
`;

exports[`<CarouselButton /> renders correct without props 1`] = `
<button
  aria-label="вперед"
  className="control"
>
  <Icon
    icon={[Function]}
    role="presentation"
    size={24}
  />
</button>
`;
